addons/mrp_account/tests/test_bom_price.py

```diff
@@ -254,3 +254,85 @@ def test_02_compute_byproduct_price(self):
         self.assertEqual(self.dining_table.standard_price, 137.5, "After computing price from BoM price should be 137.5")
         scrap_wood.button_bom_cost()
         self.assertEqual(scrap_wood.standard_price, 20.63, "After computing price from BoM price should be 20.63")
+
+
+    def test_compute_average_price_with_different_company(self):
+        """Ensure _compute_average_price correctly fetches price from the intended company"""
+
+        # Create Companies
+        self.company_a = self.env['res.company'].create({'name': 'Company A'})
+        self.company_b = self.env['res.company'].create({'name': 'Company B'})
+
+        # Create Product Category with cost method other than 'standard'
+        self.product_category = self.env['product.category'].with_company(self.company_b).create({
```

Comment:
there's no need to add `self` in front of these variables, it makes the test more confusing as `self` is used to reference variables created in the `setUpClass`